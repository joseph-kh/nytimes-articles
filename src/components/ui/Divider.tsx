import React from "react";
import { StyleSheet, View as ViewElement, ViewProps } from "react-native";
import COLORS from "@/theme/colors";
import { moderateScale } from "@/utils/scaler";

const Divider: React.FC<ViewProps> = (props) => {
  const { style } = props;

  return (
    <ViewElement
      {...props}
      style={StyleSheet.flatten([styles.seperatorContainer, style])}
    />
  );
};

export default Divider;

const styles = StyleSheet.create({
  seperatorContainer: {
    borderBottomWidth: moderateScale(0.5),
    backgroundColor: COLORS.gray,
    opacity: 0.4,
  },
});
