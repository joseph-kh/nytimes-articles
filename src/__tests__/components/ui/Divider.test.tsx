import React from "react";
import { create } from "react-test-renderer";
import Divider from "@/components/ui/Divider";

describe("Divider Component", () => {
  it("renders correctly", () => {
    const tree = create(<Divider />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies custom styles", () => {
    const customStyle = { backgroundColor: "red" };
    const tree = create(<Divider style={customStyle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
