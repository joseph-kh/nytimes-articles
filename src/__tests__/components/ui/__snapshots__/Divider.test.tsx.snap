// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component applies custom styles 1`] = `
<View
  style={
    {
      "backgroundColor": "red",
      "borderBottomWidth": 0.75,
      "opacity": 0.4,
    }
  }
/>
`;

exports[`Divider Component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#7e7e7e",
      "borderBottomWidth": 0.75,
      "opacity": 0.4,
    }
  }
/>
`;
